### PR TITLE
Migrate tag_status to new GlobalCommands module

### PR DESCRIPTION
**Pull request by t-wissmann**
The GlobalCommands module collects commands that modify the global state
of hlwm and need multiple subsystems (e.g. monitors and tags).

As an example, the tag_status command is migrated, and it's test suite
is moved to the new test_global_commands.py and slightly extended.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -155,7 +156,8 @@ unique_ptr<CommandTable> commands(shared_ptr<Root> root) {
                                      &ClientManager::fullscreen_complete}},
         {"pseudotile",     {clients, &ClientManager::pseudotile_cmd,
                                      &ClientManager::pseudotile_complete}},
-        {"tag_status",     print_tag_status_command},
+        {"tag_status",     {global_cmds, &GlobalCommands::tagStatusCommand,
+                                          &GlobalCommands::tagStatusCompletion}},
```
Comment (blarz) [inline src/main.cpp:160]: My OCD is tingling with the off-by-one indent :D

**Review by blarz — CHANGES_REQUESTED** [review]

**Comment by blarz** [comment]
Not sure what is happening but compiling this PR locally gives me this linker error:
```
/usr/bin/ld: CMakeFiles/herbstluftwm.dir/main.cpp.o: in function `std::_Bind<int (GlobalCommands::*(GlobalCommands*, std::_Placeholder<1>, std::_Placeholder<2>))(Input, std::ostream&)>::_Bind(std::_Bind<int (GlobalCommands::*(GlobalCommands*, std::_Placeholder<1>, std::_Placeholder<2>))(Input, std::ostream&)>&&)':
/usr/include/c++/9/functional:473: undefined reference to `GlobalCommands::tagStatusCommand(Input, std::ostream&)'
/usr/bin/ld: CMakeFiles/herbstluftwm.dir/main.cpp.o: in function `std::_Bind<void (GlobalCommands::*(GlobalCommands*, std::_Placeholder<1>))(Completion&)>::_Bind(std::_Bind<void (GlobalCommands::*(GlobalCommands*, std::_Placeholder<1>))(Completion&)>&&)':
/usr/include/c++/9/functional:473: undefined reference to `GlobalCommands::tagStatusCompletion(Completion&)'
/usr/bin/ld: CMakeFiles/herbstluftwm.dir/root.cpp.o: in function `std::unique_ptr<GlobalCommands, std::default_delete<GlobalCommands> > make_unique<GlobalCommands, Root&>(Root&)':
/home/simon/git/herbstluftwm/src/utils.h:94: undefined reference to `GlobalCommands::GlobalCommands(Root&)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/herbstluftwm.dir/build.make:939: herbstluftwm] Error 1
make[1]: *** [CMakeFiles/Makefile2:197: src/CMakeFiles/herbstluftwm.dir/all] Error 2
make: *** [Makefile:130: all] Error
```
with
```
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
```

**Comment by t-wissmann** [comment]
Strange.. Did you re-run cmake? Does it work in a fresh build directory?

**Comment by blarz** [comment]
> Strange.. Did you re-run cmake? Does it work in a fresh build directory?

I tried it after re-running cmake (just ```cmake ..```), still the same issue.

A fresh build dir fixed it, thanks!

File: src/main.cpp (hunk 1 above)
Comment (t-wissmann) [reply src/main.cpp]: [x] fixed :)

Comment (blarz) [reply src/main.cpp]: Thanks!

**Review by blarz — APPROVED** [review]

**Pull request merged**